### PR TITLE
Migration config implementation

### PR DESCRIPTION
**Pull request by MiguelAngelGutierrezMaya**
### Core changes

- Added a new `@packages/migrations` workspace package with a versioned migration runner for PostgreSQL.
- Implemented CLI commands to run migrations, rollback the latest version, inspect migration status, and scaffold new migration files.
- Added local Supabase support and root-level scripts to manage the local database workflow used by migrations.
- Updated the infrastructure entry config to support deploying multiple versions at the same time and removed the unused `v1` database stack.

### Schema changes

- Added migration `1.0.0` to create the base financial schema, including `documents`, `providers`, `currencies`, `expenses_types`, `expenses_categories`, `users`, `expenses`, and `audit_logs`.
- Added triggers and functions to keep `updated_at` in sync and to register automatic audit entries for inserts, updates, and deletes.
- Added migration `1.0.1` to seed initial catalog data for currencies, expense types, expense categories, providers, and document types.
- Added migration `1.1.0` to create composite indexes for the most frequent `expenses` query patterns.

### Minor changes

- Updated environment setup so database-related variables are copied into the migrations and Supabase packages during local setup.
- Added Turbo and workspace configuration for the new Supabase and migrations commands and dependencies.
- Removed the unused `@services/auth` package and cleaned up related infrastructure dependencies.
- Standardized package metadata updates across the workspace.

## Screenshots/Recordings

N/A

## How to Test

1. Run `pnpm install`.
2. Run `./config/helpers/setup_env.sh` and then `direnv allow`.
   Expected result: local environment files are created and database variables are available for the migrations package.
3. Run `pnpm supabase:start`.
   Expected result: the local Supabase stack starts successfully and Postgres is available.
4. Run `pnpm migrations:status`.
   Expected result: the database shows no current version and reports pending migrations up to `1.1.0`.
5. Run `pnpm migrations:migrate`.
   Expected result: migrations `1.0.0`, `1.0.1`, and `1.1.0` are applied successfully and the command finishes with the database migrated to `1.1.0`.
6. Run `pnpm migrations:status` again.
   Expected result: the database is reported as up to date at version `1.1.0`.
7. Run `pnpm migrations:rollback`.
   Expected result: the latest migration is rolled back successfully and the database version returns to `1.0.1`.
8. Run `pnpm --filter @packages/migrations test`.
   Expected result: the migration CLI, runner, config, and semantic version tests pass.
9. Run `pnpm infra:cdk synth`.
   Expected result: synthesis completes using the multi-version deployment config without recreating the removed `v1` database stack.

## Checklist

- [x] The PR description includes clear instructions on how to test the implementation manually.

##### Added Tests?

- [x] yes
  - [x] Verified that this PR is not creating duplicate use cases
  - [ ] All edge cases described in the acceptance criteria are covered, and tests have clear and descriptive names
- [ ] no, because they aren't needed
- [ ] no, because I need help

_Override these when needed._

```sh

```

### CODE REVIEW

File: packages/migrations/src/runner/get-version-list.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      for (const patch of readNumericDirs(minorDir)) {
+        const patchDir = path.join(minorDir, String(patch));
+        const versionFile = path.join(patchDir, 'version.ts');
+
+        if (!fs.existsSync(versionFile)) continue;
+
+        // eslint-disable-next-line @typescript-eslint/no-require-imports
+        const mod = require(versionFile);
+        const config: VersionConfig = mod.default ?? mod;
+
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply packages/migrations/src/runner/get-version-list.ts:31]: package run using tsx, that function works

File: packages/migrations/src/runner/get-execution-fn.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      case 'ts': {
+        const tsPath = path.join(versionDir, script.path);
+        // eslint-disable-next-line @typescript-eslint/no-require-imports
+        const mod = require(tsPath);
+        if (typeof mod.up !== 'function' || typeof mod.down !== 'function') {
+          throw new Error(
+            `TS migration ${tsPath} must export up() and down() functions`,
+          );
+        }
+        return { up: mod.up, down: mod.down };
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply packages/migrations/src/runner/get-execution-fn.ts:45]: Same case, tsx works

File: packages/migrations/src/config.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import type { DatabaseConfig } from './interfaces/database';
+
+export interface AppConfig {
+  db: DatabaseConfig;
+  migrationsDir: string;
+}
+
+// Package root: packages/migrations/ (one level up from src/)
+const packageRoot = path.resolve(__dirname, '..');
+
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply packages/migrations/src/config.ts:12]: tsx has _dirname, it is not a problem

File: packages/migrations/src/config.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        'Ensure DATABASE_URL is defined in your .env file or loaded via direnv.',
+    );
+  }
+
+  const schema = process.env.DATABASE_SCHEMA || 'financial_management';
+
+  return {
+    db: { connectionString, schema },
+    migrationsDir: path.resolve(__dirname, 'migrations'),
+  };
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply packages/migrations/src/config.ts:35]: Idem

File: .envrc [hunk 5]
```diff
@@ -16,40 +16,63 @@ echo "🚀 Environment: $ENV"
...
+        cp ./config/.env.development ./client/main/.env.local
 
-    echo "🚀 Loading environment variables on client/main"
-    [[ -f ./client/main/.env.development ]] && cp ./client/main/.env.development ./client/main/.env.development.local
+        echo "🚀 Copying environment variables to packages/migrations"
+        cp ./config/.env.development ./packages/migrations/.env.local
+
+        echo "🚀 Copying environment variables to packages/supabase"
+        cp ./config/.env.development ./packages/supabase/.env
+    fi
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply .envrc:59]: This file has been refactored

File: packages/migrations/src/cli.ts [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+    )
+    .demandCommand(1, 'You must specify a command')
+    .strict()
+    .help();
+}
+
+// Run when executed directly (not when imported by tests)
+if (require.main === module) {
+  buildCli(hideBin(process.argv)).parse();
+}
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply packages/migrations/src/cli.ts:334]: tsx provides require.main function

File: packages/migrations/src/runner/get-db-version.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+const ENSURE_TABLE = `
+CREATE TABLE IF NOT EXISTS schema_migrations (
+    id              uuid DEFAULT gen_random_uuid() PRIMARY KEY,
+    version         varchar(20) NOT NULL,
+    description     text,
+    executed_at     timestamptz DEFAULT now(),
+    execution_time  integer,
+    checksum        varchar(64),
+    success         boolean DEFAULT true
+);
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply packages/migrations/src/runner/get-db-version.ts:12]: supabase use PG 17, which has the function by default
